uaclient/entitlements/tests/test_livepatch.py

```diff
@@ -43,59 +43,64 @@
 M_PATH = 'uaclient.entitlements.livepatch.'  # mock path
 
 
-class BaseEnabledLivepatchConfig(TestCase):
-    """Base class which sets up an enabled livepatch entitlement."""
+@pytest.fixture
+def livepatch_entitlement(tmpdir):
+    """
+    A pytest fixture to create a LivepatchEntitlement with some default config
 
-    def setUp(self):
-        super().setUp()
-        tmp_dir = self.tmp_dir()
-        self.cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
-        self.cfg.write_cache('machine-token', dict(LIVEPATCH_MACHINE_TOKEN))
-        self.cfg.write_cache('machine-access-livepatch',
-                             dict(LIVEPATCH_RESOURCE_ENTITLED))
-        self.entitlement = LivepatchEntitlement(self.cfg)
+    (Uses the tmpdir fixture for the underlying config cache.)
+    """
+    cfg = config.UAConfig(cfg={'data_dir': tmpdir.strpath})
+    cfg.write_cache('machine-token', dict(LIVEPATCH_MACHINE_TOKEN))
+    cfg.write_cache('machine-access-livepatch',
+                    dict(LIVEPATCH_RESOURCE_ENTITLED))
+    return LivepatchEntitlement(cfg)
 
 
-class TestLivepatchContractStatus(BaseEnabledLivepatchConfig):
+class TestLivepatchContractStatus:
```

Comment:
`pytest` generally leaves unittest.TestCase derived classes alone, as the interactions between `pytest` and `unittest` could cause complex issues.

Note that this means that all the assertion methods on `self` aren't available; pytest instruments the `assert` statement to provide more useful output than most of those assertion methods anyway, so they generally aren't missed.

(This particular class doesn't use any `pytest` functionality that requires us to drop the `TestCase` inheritance, but as I was removing `BaseEnabledLivepatchConfig` I was modifying its definition anyway, so I dropped the inheritance regardless.)